### PR TITLE
Revert "Setup"

### PR DESCRIPTION
**Pull request by LionRouge1**
Reverts for a review

### CODE REVIEW

File: src/functionBased/components/TodoContainer.js [hunk 1]
```diff
@@ -1,59 +1,62 @@
...
-import Header from './Header';
-import InputTodo from './InputTodo';
-import TodoList from './TodosList';
+import React, {useState, useEffect} from "react";
+import Header from "./Header";
+import InputTodo from "./InputTodo";
+import TodoList from "./TodosList";
 
 const TodoContainer = () => {
-  function getInitialTodos() {
```
Comment (brahimdidi) [inline src/functionBased/components/TodoContainer.js:7]: please use arrow functions

Comment (brahimdidi) [inline src/functionBased/components/TodoContainer.js:7]: please use arrow functions

File: src/functionBased/pages/SinglePage.js [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
-      <h2>Another Single page</h2>
-      <button type="button" onClick={() => navigation(-1)}>Go back</button>
-    </div>
-  );
-};
-export default SinglePage;
+  const navigation = useNavigate()
+  const aboutData = [
+    {
+      slug: "about-app",
```
Comment (brahimdidi) [inline src/functionBased/pages/SinglePage.js:8]: be consistent and use single quotes or double quotes but do not mix it.

Comment (brahimdidi) [inline src/functionBased/pages/SinglePage.js:8]: be consistent and use single quotes or double quotes but do not mix it.

File: README.md [hunk 3]
```diff
@@ -17,35 +11,60 @@ In the project directory, you can run:
...
 
+The page will reload when you make changes.\
+You may also see any lint errors in the console.
 
 ### `npm test`
 
 Launches the test runner in the interactive watch mode.\
 See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
 
-## Author
```
Comment (brahimdidi) [inline README.md:26]: you did not specify the author

Comment (brahimdidi) [inline README.md:26]: you did not specify the author

File: src/classBased/components/TodoItem.js [hunk 4]
```diff
@@ -1,64 +1,64 @@
...
-      todo, handleChange, deleteTodo, setUpdate,
-    } = this.props;
-    const { completed, id, title } = todo;
+      textDecoration: "line-through",
+    }
+    const {todo, handleChange, deleteTodo, setUpdate} = this.props;
+    const {completed, id, title} = todo;
     return (
       <li className={styles.item}>
-        <div onDoubleClick={this.handleEditing} style={viewMode}>
```
Comment (brahimdidi) [inline src/classBased/components/TodoItem.js:49]: I think it is weird to have a div inside an li don't you thnk so

File: src/classBased/components/TodoItem.js [hunk 5]
```diff
@@ -1,64 +1,64 @@
...
-          <input
-            type="checkbox"
+        <div onDoubleClick={this.handleEditing} style={viewMode} >
+          <input
```
Comment (brahimdidi) [inline src/classBased/components/TodoItem.js:50]: I would suggest using a label with the input

Comment (brahimdidi) [inline src/classBased/components/TodoItem.js:50]: I would suggest using a label with the input

**Review by brahimdidi — APPROVED** [review]
Hi  @LionRouge1 ,

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉
Optional changes, Check the comments down below.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Comment by LionRouge1** [comment]
Thanks for the feedback :heart_eyes: I will see how to implement all you suggestion

**Pull request merged**